src/cbc_sdk/platform/alerts.py

```diff
@@ -1496,3 +1552,66 @@ def set_remote_is_private(self, is_private, exclude=False):
         else:
             self._exclusions["remote_is_private"] = is_private
         return self
+
+
+class GroupedAlertSearchQuery(AlertSearchQuery):
+    """Represents a query that is used to group Alert objects by a given field."""
+    def __init__(self, *args, **kwargs):
+        """Initialize the GroupAlertSearchQuery."""
+        super().__init__(*args, **kwargs)
+        self._group_by = "THREAT_ID"
```

Comment:
group by field should be an input parameter with a default value of THREAT_ID, not hard coded.

Reply:
do we need an input parameter on the init since we are defaulting to threat_id? if we use GASQ.set_group_by(FIELD) that will override the initialized THREAT_ID

Reply:
No the design is to default to `threat_id` and if the user wants to change it then they can use set_group_by. We don't recommend users to create queries directly.

```
query = cb.select(GroupedAlert)
query.set_group_by("new_field")
```